### PR TITLE
Add CI

### LINKED ISSUE #2: Create github action to run unittests
Title says it all.

Also, I already have a test.py file, which you can find [here](https://github.com/hot9cups/probdists/blob/main/test.py). So you're free to use that if you want to.
I'd like the github action to run the unittests for every push made to master branch, and for every Pull Request opened.

### PR DESCRIPTION
**Pull request by augusto-jm-amaral**
Add Lint and running tests on push.

closes #2 #5

### CODE REVIEW

**Comment by augusto-jm-amaral** [comment]
Here is the running https://github.com/augusto-jm-amaral/probdists/runs/1309028325?check_suite_focus=true, have one test that failed, I don`t know how to solve yet 🤔

**Comment by hot9cups** [comment]
@augusto-jm-amaral Most of it looks great!
I will have to check why the last test isn't passing, it was working fine the last I checked so I'll have to see if it's a problem on my side or if there's a bug in the script. I'll let you know once I get access to my pc, which most probably will be tomorrow.

**Comment by hot9cups** [comment]
@augusto-jm-amaral I got access to my pc and was able to troubleshoot.
The problem was that there was a missing return statement in the replace_data_with_stats() function, which was why the test was failing. I've gone ahead and pushed the fix to that, kindly fork the latest version of the repo and let me know if everything's okay now :)

**Comment by hot9cups** [comment]
Wanted to update, the tests with Exponential Distribution unittests now work correctly
image

**Comment by augusto-jm-amaral** [comment]
@hot9cups Nice, now works, [look here](https://github.com/augusto-jm-amaral/probdists/runs/1310529040?check_suite_focus=true), I had to add the test files in the repository, I don`t know if it`s ok

**Comment by hot9cups** [comment]
@augusto-jm-amaral "I had to add the test file in the repository". 

I didn't understand, isn't the test.py already in the repository? What do you mean?

**Comment by augusto-jm-amaral** [comment]
The sample files, used in tests, I have to add for running the test.py.

**Comment by hot9cups** [comment]
@augusto-jm-amaral Thanks for bringing that to my notice. I have changed the logic to obtain the data files in the code, now you shouldn't need to add them in manually. Let me know if everything works fine on the latest fork of the repo :)

**Comment by augusto-jm-amaral** [comment]
@hot9cups works fine https://github.com/augusto-jm-amaral/probdists/runs/1318229700?check_suite_focus=true, thanks 🥳

**Review by hot9cups — COMMENTED** [review]
These are last pending bits, and then the PR is ready to be merged :)

File: .github/workflows/pullrequest.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8 pytest
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+    - name: Lint with flake8
+      run: |
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
```
Comment (hot9cups) [inline .github/workflows/pullrequest.yml:29]: I would also like --ignore=E203,W503 to be added here

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+probdists==1.9
```
[the remark this answers is not in the document]
Comment (augusto-jm-amaral) [reply requirements.txt:1]: https://github.com/pagarme/cardoso/pull/156

File: .github/workflows/pullrequest.yml (hunk 1 above)
Comment (augusto-jm-amaral) [reply .github/workflows/pullrequest.yml]: Done

File: .github/workflows/pullrequest.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+on: [push]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.7]
```
[the remark this answers is not in the document]
Comment (augusto-jm-amaral) [reply .github/workflows/pullrequest.yml]: Done

File: .github/workflows/pullrequest.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+name: Python Lint and Test
+
+on: [push]
```
[the remark this answers is not in the document]
Comment (augusto-jm-amaral) [reply .github/workflows/pullrequest.yml]: Done

File: .github/workflows/pullrequest.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
+name: Python Lint and Test
+
+on: [pull_request]
```
Comment (hot9cups) [inline .github/workflows/pullrequest.yml:3]: heh, I think there's a misunderstanding. I said both push and pull_request :D

File: requirements.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (augusto-jm-amaral) [reply requirements.txt:1]: Yes, for load the dependencies, it's a pattern, I use `pip freeze > requirements.txt` to lock the dependencies and `pip install -r requirements.txt` install that.

File: .github/workflows/pullrequest.yml (hunk 5 above)
Comment (augusto-jm-amaral) [reply .github/workflows/pullrequest.yml]: Done @hot9cups

**Review by hot9cups — APPROVED** [review]
Wonderful!
You have been really diligent and precise with this PR. I can barely fault anything now, thank you so much for going back and forth with me and improving this till it was perfect! After all those iterations, this is now brilliant!

Thanks a lot for your contribution! I hope you find this library useful in your work in the future!
Also, is it fine with you if I include you as a contributor when I release the next version of the library? 
:)

**Comment by augusto-jm-amaral** [comment]
Thank you @hot9cup, I'm not familiar with python but I learn a lot with this PR, it looks great to me if you add me as a contributor 🎉

**Comment by hot9cups** [comment]
You've done great work @augusto-jm-amaral ! I'm glad this PR helped you learn tons :)
I'll notify you when I release the version! Thank you once again :)

**Pull request merged**